Source/Urho3D/Scene/Node.cpp

```diff
@@ -1634,6 +1634,60 @@ const Variant& Node::GetVarByHash(StringHash key) const
     return i != vars_.end() ? i->second : Variant::EMPTY;
 }
 
+void Node::GetDerivedComponents(ea::vector<Component*>& dest, StringHash type, bool recursive) const
+{
+    dest.clear();
+
+    if (!recursive)
+    {
+        for (auto i = components_.begin(); i != components_.end(); ++i)
+        {
+            if ((*i)->GetTypeInfo()->IsTypeOf(type))
+                dest.push_back(i->Get());
+        }
+    }
+    else
+        GetDerivedComponentsRecursive(dest, type);
```

Comment:
Why is `GetDerivedComponentsRecursive` a thing? We could just do this:
```cpp
void Node::GetDerivedComponents(ea::vector<Component*>& dest, StringHash type, bool recursive) const
{
    for (auto i = components_.begin(); i != components_.end(); ++i)
    {
        if ((*i)->GetTypeInfo()->IsTypeOf(type))
            dest.push_back(i->Get());
    }
    
    if (recursive)
    {
        for (auto i = children_.begin(); i != children_.end(); ++i)
            (*i)->GetDerivedComponents(dest, type, recursive);
    }
}
```

Reply:
I don't know, I've copied the current code and modified it to return derived components. Shall I change both? Can I keep it as-is?

Reply:
Ah.. That myth clean Urho3D codebase 😁 
Do this bit if cleanup if you don't mind 🙏